packages/web/e2e/signUp.test.ts

```diff
@@ -0,0 +1,277 @@
+import path from 'path'
+
+import { Page, expect, devices } from '@playwright/test'
+
+import { test } from './test'
+import { resetAuthState, generateTestUser } from './utils'
+
+type User = {
+  email: string
+}
+
+async function expectSignUpPage(page: Page) {
+  await page.waitForURL(/signup/i)
+  await expect(
+    page.getByRole('heading', { name: /sign up for audius/i, level: 1 })
+  ).toBeVisible()
+}
+
+async function assertOnCreatePasswordPage(user: User, page: Page) {
+  await expect(
+    page.getByRole('heading', { name: /create your password/i })
+  ).toBeVisible()
+
+  await expect(
+    page.getByText(/Create a password that’s secure and easy to remember!/i)
+  ).toBeVisible()
+
+  await expect(page.getByText(/your email/i)).toBeVisible()
+  await expect(page.getByText(user.email)).toBeVisible()
+}
+
+async function testSignUp({
+  isMobile = false,
+  testUser = generateTestUser(),
+  signUpUrl = 'signup',
+  page
+}: {
+  isMobile?: boolean
+  testUser?: ReturnType<typeof generateTestUser>
+  signUpUrl?: string
+  page: Page
+}) {
+  const { email, password, handle, name } = testUser
+
+  // --> Load signup route
+  await page.goto(signUpUrl)
+  await expect(
+    page.getByRole('heading', { name: /sign up for audius/i, level: 1 })
+  ).toBeVisible({ timeout: 45000 })
+
+  await page.getByRole('button', { name: /sign up free/i }).click()
+
+  // --> Email Page
+  await page.getByRole('textbox', { name: /email/i }).fill(email)
+  await page.getByRole('button', { name: /sign up free/i }).click()
+
+  await assertOnCreatePasswordPage(testUser, page)
+
+  // --> Password Page
+  // Password inputs dont have a role, so we just check against label text
+  // https://github.com/testing-library/dom-testing-library/issues/567#issue-616906804
+  await page.getByLabel(/^password/i).fill(password)
+  await page.getByLabel(/confirm password/i).fill(password)
+
+  await page.getByRole('button', { name: /continue/i }).click()
+
+  // --> Pick Handle Page
+  await expect(
+    page.getByRole('heading', { name: /pick your handle/i })
+  ).toBeVisible()
+  await expect(
+    page.getByText(/this is how others find and tag you/i)
+  ).toBeVisible()
+  await page.getByRole('textbox', { name: /handle/i }).fill(handle)
+  await page.getByRole('button', { name: /continue/i }).click()
+
+  // --> Finish Profile Page
+  await expect(
+    page.getByRole('heading', { name: /finish your profile/i })
+  ).toBeVisible()
+  await expect(
+    page.getByText(/your photos & display name is how others see you./i)
+  ).toBeVisible()
+
+  // upload cover & profile photo
+
+  // TODO: whats the best file upload method?
```

Comment:
I definitely like the `setInputFiles` method the most, but I think in some cases we'll want to make sure eg drag and drop works